pymc3/variational/updates.py

```diff
@@ -0,0 +1,150 @@
+###########################################################
+#  Refactored by Dilin Wang
+#  Original implementation by Alec Radford, Luke Metz, Soumith Chintala: https://github.com/Newmu/dcgan_code   
+#  (c) Alec Radford, Luke Metz, Soumith Chintala
+###########################################################
+
+import theano
+import theano.tensor as tt
+import numpy as np
+
+
+__all__ = [
+    "SGD",
+    "Momentum",
+    "Adagrad",
+    "RMSprop",
+    "Adadelta",
+    "Adam"
+]
+
+def floatX(X):
+    return np.asarray(X, dtype=theano.config.floatX)
+
+class Update(object):
+
+    def __init__(self):
+        self.__dict__.update(locals())
+
+    def __call__(self, params, grads):
+        raise NotImplementedError
+
+
+class SGD(Update):
+
+    def __init__(self, lr=0.01, *args, **kwargs):
+        Update.__init__(self, *args, **kwargs)
+        self.__dict__.update(locals())
```

Comment:
You update dict twice, moreover passing any additional argument will cause an error, but signature allows it (e.g. `SGD(foo=1)` will fail, but it is not obvious)